clinica/pipelines/deeplearning_prepare_data/deeplearning_prepare_data_utils.py

```diff
@@ -0,0 +1,228 @@
+    :param preprocessed_T1:
+    :return:
+    """
+    import torch
+    import os
+
+    basedir = os.getcwd()
+    image_tensor = torch.load(preprocessed_T1)
+
+    # use classifiers tensor.upfold to crop the patch.
+    patches_tensor = image_tensor.unfold(1, patch_size, stride_size).unfold(2, patch_size, stride_size).unfold(3, patch_size, stride_size).contiguous()
+    # the dimension of patch_tensor should be [1, patch_num1, patch_num2, patch_num3, patch_size1, patch_size2, patch_size3]
+    patches_tensor = patches_tensor.view(-1, patch_size, patch_size, patch_size)
+
+    output_patch = []
+    for index_patch in range(patches_tensor.shape[0]):
+        extracted_patch = patches_tensor[index_patch, ...].unsqueeze_(0)  # add one dimension
+        # save into .pt format
+        output_patch.append(
+                os.path.join(
+                    basedir,
+                    os.path.basename(preprocessed_T1).split('.pt')[0]
+                    + '_patchsize-'
+                    + str(patch_size)
+                    + '_stride-'
+                    + str(stride_size)
+                    + '_patch-'
+                    + str(index_patch)
+                    + '.pt'
+                    )
+                )
+        torch.save(extracted_patch.clone(), output_patch[index_patch])
+
+    return output_patch
+
+
+def save_as_pt(input_img):
+    """
+    This function transforms  nii.gz file into .pt format, in order to train
+    the classifiers model more efficient when loading the data.
+    :param input_img:
+    :return:
+    """
+
+    import torch
+    import os
+    import nibabel as nib
+
+    basedir = os.getcwd()
+    image_array = nib.load(input_img).get_fdata()
+    image_tensor = torch.from_numpy(image_array).unsqueeze(0).float()
+    # make sure the tensor dtype is torch.float32
+    output_file = os.path.join(basedir, os.path.basename(input_img).split('.nii.gz')[0] + '.pt')
+    # save
+    torch.save(image_tensor.clone(), output_file)
+
+    return output_file
+
+
+def get_data_datasink(image_id):
+    substitutions_ls = [  # registration
+            (image_id + '_T1w_corrected.nii.gz',
+                image_id + '_T1w_desc-BiasCorrected_T1w.nii.gz'),
+            (image_id + 'Warped_cropped_intensity_norm.nii.gz',
+                image_id + '_T1w_space-MNI152NLin2009cSym_res-1x1x1_intensity_norm_T1w.nii.gz'),
+            (image_id + 'Warped_cropped.nii.gz',
+                image_id + '_T1w_space-MNI152NLin2009cSym_desc-Crop_res-1x1x1_T1w.nii.gz'),
+            (image_id + '0GenericAffine.mat',
+                image_id + '_T1w_space-MNI152NLin2009cSym_res-1x1x1_affine.mat'),
+            (image_id + 'Warped_cropped.pt',
+                image_id + '_T1w_space-MNI152NLin2009cSym_res-1x1x1_T1w.pt'),
+            (image_id + 'Warped.nii.gz',
+                image_id + '_T1w_space-MNI152NLin2009cSym_res-1x1x1_T1w.nii.gz')
+            ]
+    return image_id, substitutions_ls
```

Comment:
My current understanding is that this is not needed anymore:

```suggestion

```

Reply:
Yes, you're right. No substitutions in this pipeline. But before removing I have to modify the nodes connection.

Reply:
I just proposed a modification of the `build_output_node` method :)